src/coreclr/vm/gccover.cpp

```diff
@@ -1816,37 +1816,19 @@ void DoGcStress (PCONTEXT regs, NativeCodeVersion nativeCodeVersion)
         frame.Push(pThread);
     }
 
-    DWORD_PTR retValRegs[2] = { 0 };
-    UINT  numberOfRegs = 0;
+    // The legacy X86 GC encoder does not encode the state of return registers at
+    // call sites, so we must add an extra frame to protect returns.
+#ifdef TARGET_X86
+    DWORD_PTR retValReg = 0;
 
     if (afterCallProtect[0])
```

Comment:
Given that we only need `afterCallProtect` for x86 now, we can also delete the whole bunch of arch-specific code above to compute it for non-x86 platforms.

Reply:
There is a lot of cleanup possible since we’ve incremented r2r version. 

I’d like to do that in a separate follow ip change as that could be larger, but mostly mechanical and unlikely to break anything change.

Reply:
We would not need disassemble anything other than x86. 
Perhaps x64 will merge with RISC and only x86 will be special.

Reply:
For non-x86 it will be just iterating over interruptible locations (partial or full interruptible - just different iterators) and stick HLT there.

Reply:
Although that may still require disassembling - so that hlt is on instruction edges.